storm-core/src/clj/backtype/storm/daemon/nimbus.clj

```diff
@@ -130,22 +162,40 @@
 (defn inbox [nimbus]
   (master-inbox (:conf nimbus)))
 
-(defn- read-storm-conf [conf storm-id]
-  (let [stormroot (master-stormdist-root conf storm-id)]
-    (merge conf
-       (clojurify-structure
-         (Utils/fromCompressedJsonConf
-           (FileUtils/readFileToByteArray
-             (File. (master-stormconf-path stormroot))))))))
+(defn- get-subject []
+  (let [req (ReqContext/context)]
+    (.subject req)))
+
+(defn- read-storm-conf [conf storm-id blob-store]
+  (clojurify-structure
+    (Utils/fromCompressedJsonConf
+      (.readBlob blob-store (master-stormconf-key storm-id) (get-subject)))))
 
 (declare delay-event)
 (declare mk-assignments)
 
+(defn nimbus-subject
+  []
+  (let [subject (Subject.)
+        principal (NimbusPrincipal.)
+        principals (.getPrincipals subject)]
+    (.add principals principal)
+    subject))
+
+(def get-nimbus-subject
+  (nimbus-subject))
```

Comment:
Looks like the var `get-nimbus-subject` and the function above `nimbus-subject` it have swapped names.  Can we switch them?